### ISSUE TITLE
Mallet operator inside async function

### ISSUE THREAD

**Issue by valeriangalliat** [babel/babel#874]
Hello,

I'm encountering a bug with playground mallet operator `||=` inside async functions.

Babel version: 4.4.5

Input:

```js
const x = async a => a.b ||= 1
// const x = async a => a.b = a.b || 1 // this one works

x({}).then(console.log, err => setTimeout(() => { throw err }))
```

Run:

```sh
babel-node --experimental --playground test.js
```

Output:

```
const x = async a => a.b ||= 1
                 ^
ReferenceError: _a is not defined
```

Compiled:

```js
"use strict";

var _this = this;

var x = function callee$0$0(a) {
  return regeneratorRuntime.async(function callee$0$0$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        return context$1$0.abrupt("return", (_a = a, !_a.b && (_a.b = 1), _a.b));

      case 1:
      case "end":
        return context$1$0.stop();
    }
  }, null, _this);
};

x({}).then(console.log, function (err) {
  return setTimeout(function () {
    throw err;
  });
});
```

**Comment by sebmck** [comment]
Looks like the scope temp variables are being lost somewhere.

**Comment by valeriangalliat** [comment]
You rock! Thanks for the quick fix. :smile:

**Comment by sebmck** [comment]
Fixed as of 4.4.6, thanks!

**Comment by DenisIzmaylov** [comment]
I got the same troubles during execute compiled code.
Original code:
```javascript
  router.get('/', function *() {
    yield this.render('get-service/landing-page', {
      title: 'Great Landing'
    });
  });
```

Final code:
```javascript
  router.get('/', regeneratorRuntime.mark(function callee$0$0() {
    return regeneratorRuntime.wrap(function callee$0$0$(context$1$0) {
      while (1) switch (context$1$0.prev = context$1$0.next) {
        case 0:
          context$1$0.next = 2;
          return this.render('get-service/landing-page', {
            title: 'Great Landing'
          });

        case 2:
        case 'end':
          return context$1$0.stop();
      }
    }, callee$0$0, this);
  }));
```

The last code causes error:
```
  ReferenceError: callee$0$0 is not defined
```
